components/clp-package-utils/clp_package_utils/scripts/native/search.py

```diff
@@ -75,79 +72,30 @@ def process_error_callback(err):
-                                 search_controller_host: str, search_controller_port: int):
+                                 end_timestamp: int | None, path_filter: str):
     search_config = SearchConfig(
-        search_controller_host=search_controller_host,
-        search_controller_port=search_controller_port,
-        wildcard_query=wildcard_query,
+        query_string=wildcard_query,
         begin_timestamp=begin_timestamp,
         end_timestamp=end_timestamp,
         path_filter=path_filter
     )
 
     sql_adapter = SQL_Adapter(db_config)
-    zstd_cctx = zstandard.ZstdCompressor(level=3)
     with closing(sql_adapter.create_connection(True)) as \
             db_conn, closing(db_conn.cursor(dictionary=True)) as db_cursor:
         # Create job
-        db_cursor.execute(f"INSERT INTO `search_jobs` (`search_config`) VALUES (%s)",
-                          (zstd_cctx.compress(msgpack.packb(search_config.dict())),))
+        db_cursor.execute(f"INSERT INTO `{SEARCH_JOBS_TABLE_NAME}` (`search_config`) VALUES (%s)",
+                          (msgpack.packb(search_config.dict()),))
         db_conn.commit()
         job_id = db_cursor.lastrowid
 
-        next_pagination_id = 0
-        pagination_limit = 64
-        num_tasks_added = 0
-        query_base_conditions = []
-        if begin_timestamp is not None:
-            query_base_conditions.append(f"`end_timestamp` >= {begin_timestamp}")
-        if end_timestamp is not None:
-            query_base_conditions.append(f"`begin_timestamp` <= {end_timestamp}")
-        while True:
-            # Get next `limit` rows
-            query_conditions = query_base_conditions + [f"`pagination_id` >= {next_pagination_id}"]
-            query = f"""
-                SELECT `id` FROM {CLP_METADATA_TABLE_PREFIX}archives 
-                WHERE {" AND ".join(query_conditions)} 
-                LIMIT {pagination_limit}
-                """
-            db_cursor.execute(query)
-            rows = db_cursor.fetchall()
-            if len(rows) == 0:
-                break
-
-            # Insert tasks
-            db_cursor.execute(f"""
-            INSERT INTO `search_tasks` (`job_id`, `archive_id`, `scheduled_time`) 
-            VALUES ({"), (".join(f"{job_id}, '{row['id']}', '{datetime.datetime.utcnow()}'" for row in rows)})
-            """)
-            db_conn.commit()
-            num_tasks_added += len(rows)
-
-            if len(rows) < pagination_limit:
-                # Less than limit rows returned, so there are no more rows
-                break
-            next_pagination_id += pagination_limit
-
-        # Mark job as scheduled
-        db_cursor.execute(f"""
-        UPDATE `search_jobs`
-        SET num_tasks={num_tasks_added}, status = '{JobStatus.SCHEDULED}'
-        WHERE id = {job_id}
-        """)
-        db_conn.commit()
-
         # Wait for the job to be marked complete
         job_complete = False
```

Comment:
```python
        while True:
            db_cursor.execute(f"SELECT `status` FROM `{SEARCH_JOBS_TABLE_NAME}` WHERE `id` = {job_id}")
            # There will only ever be one row since it's impossible to have more than one job with the same ID
            new_status = db_cursor.fetchall()[0]['status']
            db_conn.commit()
            if new_status in (JobStatus.SUCCESS, JobStatus.FAILED, JobStatus.CANCELLED):
                break

            time.sleep(0.5)
```